strawberry_django/auth/mutations.py

```diff
@@ -1,16 +1,25 @@
 from __future__ import annotations
 
+import contextlib
 import functools
 from typing import TYPE_CHECKING, Any, Type, cast
 
 import strawberry
+from asgiref.sync import async_to_sync
 from django.contrib import auth
 from django.contrib.auth.password_validation import validate_password
 
+from strawberry_django.auth.utils import get_current_user
 from strawberry_django.mutations import mutations, resolvers
 from strawberry_django.mutations.fields import DjangoCreateMutation
 from strawberry_django.optimizer import DjangoOptimizerExtension
 from strawberry_django.resolvers import django_resolver
+from strawberry_django.utils.requests import get_request
+
+with contextlib.suppress(ModuleNotFoundError):
+    # Django-channels is not always used/intalled,
+    # therefore it shouldn't be it a hard requirement.
+    from channels import auth as channels_auth
```

Comment:
Probably use `ImportError` and a simple `try`/`except` to assign `channels_auto = None` on errors, otherwise the name will not be defined below

Reply:
hmm, That was my original approach, but then ruff complains.

Reply:
> hmm, That was my original approach, but then ruff complains.

You can `# noqa: XYZ` on what ruff complains in that case =P

Search for `ImportError` in the repo, there are some similar approaches in it

Reply:
Interesting that setting the missing import to None seems to get ignored by ruff.  

So I picked `ModuleNotFoundError` over `ImportError` since in this case we're talking about a package not being installed rather then struggling to import https://docs.python.org/3/library/exceptions.html#ModuleNotFoundError